### PR TITLE
Validate functions

### PR DESCRIPTION
**Pull request by ttuegel**
Reapply #2168.

It looks like we forgot `\or` patterns in function rules.

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/src/Kore/Equation/Validate.hs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                | TermLike.isConstructorLike term -> descend
+            TermLike.App_ sym _
+                | isGoodSymbol sym -> descend
+                | otherwise -> Just term
+            TermLike.InternalBytes_ _ _ -> Nothing
+            TermLike.InternalBool_ _ -> Nothing
+            TermLike.InternalInt_ _ -> Nothing
+            TermLike.InternalString_ _ -> Nothing
+            TermLike.DV_ _ (TermLike.StringLiteral_ _) -> Nothing
+            TermLike.And_ _ _ _ -> descend
```
Comment (MirceaS) [inline kore/src/Kore/Equation/Validate.hs:154]:
```suggestion
            TermLike.And_ _ _ _ -> descend
            TermLike.Or_ _ _ _ -> descend
```

**Comment by MirceaS** [comment]
@ttuegel You mean like this?

**Comment by MirceaS** [comment]
Tried rerunning Jenkins job: https://office.runtimeverification.com/jenkins/blue/organizations/jenkins/haskell-backend/detail/PR-2585/4/pipeline

**Comment by ttuegel** [comment]
This error will be fixed by #2598.

**Review by MirceaS — APPROVED** [review]

**Pull request merged**